reinforcement-learning-games/zombsole-gym-example/dqn/q_learning.py

```diff
@@ -0,0 +1,184 @@
+                frame = new_frame
+            
+            for _ in range(self.config['T']):
+                num_of_trials += 1
+                epsilon_greedy = self.epsilon_min + \
+                    max(self.epsilon_decay - num_of_trials, 0) / \
+                    self.epsilon_decay * (1 - self.epsilon_min)
+                if self.verbose:
+                    print("epi {}, frame {}k: reward {}, eps {}".format(episode, 
+                                                                        int(num_of_trials / 1000), 
+                                                                        total_reward,
+                                                                        epsilon_greedy))
+                if num_of_trials % self.update_interval == 0:
+                    self.optimizer.train_one_step(sess, num_of_trials, self.batch_size)
+                
+                state = self.replay_memory.phi(frame)
+                action_idx = self.choose_action(sess, state, epsilon_greedy)
+                r, new_frame, termination = self.play(action_idx)
+                total_reward += r
+                self.replay_memory.add(frame, action_idx, r, termination)
+                frame = new_frame
+                
+                # Perhaps added (num_of_trials > 0)
+                if num_of_trials % self.time_between_two_copies == 0:
+                    self.update_target_network(sess)
+                    self.save(sess, saver)
+                
+                if self.callback:
+                    self.callback()
+                if termination:
+                    score = total_reward # self.env.get_total_reward()
+                    summary_str = sess.run(self.summary_op, feed_dict={self.t_score: score})
+                    self.summary_writer.add_summary(summary_str, num_of_trials)
+                    self.summary_writer.flush()
+                    break
+    
+    def evaluate(self, sess):
+        
+        for episode in range(self.n_episode):
+            total_reward = 0
+            frame = self.env.reset()
+            # self.env.get_current_feedback()
+            for _ in range(self.num_nullops):
+                action_idx=5
+                r, new_frame, termination = self.play(action_idx) # 2024-12-18 NOTE: Changed this from ...(action=action_idx)
+                total_reward += r
+                self.replay_memory.add(frame, 0, r, termination)
+                frame = new_frame
+            
+            for _ in range(self.config['T']):
+                if self.verbose:
+                    print("episode {}, total reward {}".format(episode, 
+                                                            total_reward))
+                
+                state = self.replay_memory.phi(frame)
+                action = self.choose_action(sess, state, self.epsilon_min)     
+                r, new_frame, termination = self.play(action)
+                self.replay_memory.add(frame, action, r, termination)
```

Comment:
Forgot to add the incremental reward `r` to `total_reward` here, i.e. add a line
```
total_reward += r
```
here.